lib/docker-compose/models/compose.rb

```diff
@@ -29,6 +34,16 @@ def get_containers_by(params)
   end
 
   #
+  # Select containers based on its given name
+  # (ignore basename)
+  #
+  def get_containers_by_given_name(given_name)
+    @containers.select { |label, container|
+      container.attributes[:name].match(/\w*_#{given_name}_\d+/)
```

Comment:
I'd say it would be safer if `@dir_name` from `ComposeUtils` could be used there instead of `\w+`, no? While the chance is small it is still there that another container could coincidentally use the same naming convention.